src/pages/index.js

```diff
@@ -1,98 +1,228 @@
+    );
+    return card.generateCard();
+  }
 
-function createCard(data) {
-  const card = new Card(data, cardTemplateSelector, () => {
-    imagePopup.open(data);
+  const cardSection = new Section(
+    {
+      items: [],
+      renderer: (item) => {
+        const cardElement = createCard(item);
+        cardSection.addItem(cardElement);
+      },
+    },
+    containerSelector
+  );
+
+  function renderAllCards(apiCards) {
+    if (apiCards.length === 0) {
+      console.log("No cards to display.");
+    } else {
+      cardSection.renderItems(apiCards);
+    }
+  }
+
+  Promise.all([api.getUserInfo(), api.getInitialCards()])
+    .then(([userData, apiCards]) => {
+      userInfo.setUserInfo({
+        name: userData.name,
+        job: userData.about,
+        avatar: userData.avatar,
+      });
+      renderAllCards(apiCards);
+    })
+    .catch((err) => {
+      console.error(`Error loading user info or cards: ${err}`);
+    });
+
+  const profilePopup = new PopupWithForm("#profile-edit-modal", {
+    handleFormSubmit: (formData) => {
+      profilePopup.updateSubmitButtonText("Saving...");
+      api
+        .setUserInfo({
+          name: formData.name,
+          about: formData.about,
+        })
+        .then((userData) => {
+          userInfo.setUserInfo({
+            name: userData.name,
+            job: userData.about,
+            avatar: userData.avatar,
+          });
+          profilePopup.close();
+        })
+        .catch((err) => {
+          console.error(`Error updating profile: ${err}`);
+        })
+        .finally(() => {
+          profilePopup.updateSubmitButtonText("Save");
+        });
+    },
   });
-  return card.generateCard();
-}
-
-const cardSection = new Section(
-  {
-    items: initialCards,
-    renderer: (item) => {
-      const cardElement = createCard(item);
-      cardSection.addItem(cardElement);
+  profilePopup.setEventListeners();
+
+  const cardPopup = new PopupWithForm("#add-place-modal", {
+    handleFormSubmit: (formData) => {
+      cardPopup.updateSubmitButtonText("Saving...");
+      api
+        .addNewCard(formData)
+        .then((cardData) => {
+          const cardElement = createCard(cardData);
+          cardSection.addItem(cardElement);
+          cardPopup.close();
+        })
+        .catch((err) => {
+          console.error(`Error adding new card: ${err}`);
+        })
+        .finally(() => {
+          cardPopup.updateSubmitButtonText("Save");
+        });
     },
-  },
-  containerSelector
-);
+  });
+  cardPopup.setEventListeners();
 
-cardSection.renderItems();
+  const avatarPopup = new PopupWithForm("#avatar-edit-modal", {
+    handleFormSubmit: (formData) => {
+      avatarPopup.updateSubmitButtonText("Saving...");
+      api
+        .setUserAvatar(formData)
+        .then((userData) => {
+          userInfo.setUserInfo({
+            name: userInfo.getUserInfo().name,
+            job: userInfo.getUserInfo().job,
+            avatar: userData.avatar,
```

Comment:
You don't need to call `userInfo.getUserInfo()` 2 times for getting the data, because you have it inside `userData`

```js
        userInfo.setUserInfo({
          name: userData.name,
          job: userData.about,
          avatar: userData.avatar,
        });
```